ecosystem_cicd_tools/validations.py

```diff
@@ -246,13 +236,37 @@ def validate_plugin_version(plugin_directory=None, changelog='CHANGELOG.txt'):
 
     plugins_asset = get_plugins(plugin_directory)
     version = get_version_py(plugin_directory)
-    check_version_plugins(plugin_directory, plugins_asset, version)
+
+    check_version_plugins_and_update(plugin_directory, plugins_asset, version)
+
     check_changelog_version(version, os.path.join(plugin_directory, changelog))
     logging.info('The official version of this plugin is {version}'
                  .format(version=version))
     return version
 
 
+def check_version_plugins_and_update(path, plugins, version):
+    path_plugin = os.path.join(os.path.abspath(path))
+    for file_name in plugins:
+        version_in_plugin = get_version_in_plugin(path_plugin, file_name)
+        if version_in_plugin != version:
+            edit_version_in_plugin_yaml(path_plugin, file_name, version)
+
+
+def edit_version_in_plugin_yaml(rel_file, file_name, version):
+    logging.info('Update version in {}'.format(file_name))
+
+    with open(os.path.join(rel_file, file_name), 'r') as f:
+        file_contents = f.read()
```

Comment:
Another way to do this would be to look over lines. I don't think we need to keep the whole file in memory. This can be particularly hard in AWS plugin.

So, here's a suggestion:

```suggestion
        lines = f.readlines()
        c = 0
        for line in lines:
            c += 1
            if regex.search(line):
                break
        lines[c] = re.sub(regex, replacement, lines[c])

```

Then down in line 267 you will need to do `fp.writelines(lines)`.